src/components/Shared/SelectedRevisionsTableRow.tsx

```diff
@@ -15,50 +18,69 @@ import {
   getTreeherderURL,
   truncateHash,
 } from '../../utils/helpers';
+import RevisionSearch from './RevisionSearch';
 
 function SelectedRevisionsTableRow(props: SelectedRevisionsRowProps) {
   const { row, index, view } = props;
+  const [isOpen, setIsOpen] = useState(false);
   const dispatch = useDispatch();
+
   const commitMessage = getLatestCommitMessage(row);
   const date = formatDate(row.push_timestamp);
   const hash = truncateHash(row.revision);
   const repository = repoMap[row.repository_id] as Repository['name'];
   const treeherderURL = getTreeherderURL(row.revision, repository);
 
+  const anchorEl = document.getElementById(row.revision);
+
   return (
-    <TableRow key={row.id}>
-      <TableCell>
-        <div className="cellStyle">{index === 0 ? 'BASE' : 'NEW'}</div>
-      </TableCell>
-      <TableCell>{repository}</TableCell>
-      <TableCell>
-        <Link href={treeherderURL}>{hash}</Link>
-      </TableCell>
-      <TableCell>{row.author}</TableCell>
-      <TableCell>
-        <div className="commit-message">{commitMessage}</div>
-      </TableCell>
-      <TableCell>{date}</TableCell>
-      <TableCell>
-        {view == 'search' && (
-          <IconButton
-            id="close-button"
-            onClick={() => dispatch(deleteRevision(row.id))}
-          >
-            <Close />
-          </IconButton>
-        )}
-        {view == 'compare-results' && (
-          <IconButton
-            id={`edit-revision-button-${index}`}
-            aria-label={`edit-revision-${row.id}`}
-          >
-            {/* TODO: implement edit revision */}
-            <EditIcon />
-          </IconButton>
-        )}
-      </TableCell>
-    </TableRow>
+    <>
+      <TableRow key={row.id} id={row.revision}>
+        <TableCell>
+          <div className="cellStyle">{index === 0 ? 'BASE' : 'NEW'}</div>
+        </TableCell>
+        <TableCell>{repository}</TableCell>
+        <TableCell>
+          <Link href={treeherderURL}>{hash}</Link>
+        </TableCell>
+        <TableCell>{row.author}</TableCell>
+        <TableCell>
+          <div className="commit-message">{commitMessage}</div>
+        </TableCell>
+        <TableCell>{date}</TableCell>
+        <TableCell>
+          {view == 'search' && (
+            <IconButton
+              id="close-button"
+              onClick={() => dispatch(deleteRevision(row.id))}
+            >
+              <Close />
+            </IconButton>
+          )}
+          {view == 'compare-results' && (
+            <IconButton
+              id={`edit-revision-button-${index}`}
+              aria-label={`edit-revision-${row.id}`}
+              onClick={() => setIsOpen(true)}
+            >
+              <EditIcon />
+            </IconButton>
+          )}
```

Comment:
same idea here: instead of the `view`, could pass the icon as a parameter, as well as the onClick event handler. And in that case the `popover` would be in the component above.
This works if src/components/Shared/SelectedRevisionsTable.tsx is duplicated, I believe this would be OK because it's a small component.

Generally I believe that as engineers we sometimes have a tendancy to not duplicate, as we were often told that duplication is bad. But it can still be the best strategy. Better engineers than myself say that 2 uses isn't a duplication yet, duplication happens with 3 or more. So I think I'd do this: duplicating SelectedRevisionsTable for the 2 use cases.